airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java

```diff
@@ -0,0 +1,446 @@
+        "bool_col", false,
+        "tiny_int_one_col", false));
+    ((ObjectNode) record2).put("tiny_int_two_col", (short) 90);
+    ImmutableList<JsonNode> records = ImmutableList.of(record1, record2);
+    Set<JsonNode> originalData = new HashSet<>(records);
+    setupForComparisonBetweenFullRefreshAndCDCSnapshot(records);
+
+    AirbyteCatalog discover = source.discover(config);
+    List<AirbyteStream> streams = discover.getStreams();
+
+    assertEquals(streams.size(), 1);
+    JsonNode jsonSchema = streams.get(0).getJsonSchema().get("properties");
+    assertEquals(jsonSchema.get("id").get("type").asText(), "number");
+    assertEquals(jsonSchema.get("bool_col").get("type").asText(), "boolean");
+    assertEquals(jsonSchema.get("tiny_int_one_col").get("type").asText(), "boolean");
+    assertEquals(jsonSchema.get("tiny_int_two_col").get("type").asText(), "number");
+
+    AirbyteCatalog catalog = new AirbyteCatalog().withStreams(streams);
+    final ConfiguredAirbyteCatalog configuredCatalog = CatalogHelpers
+        .toDefaultConfiguredCatalog(catalog);
+    configuredCatalog.getStreams().forEach(c -> c.setSyncMode(SyncMode.FULL_REFRESH));
+
+    Set<JsonNode> dataFromFullRefresh = extractRecordMessages(
+        AutoCloseableIterators.toListAndClose(source.read(config, configuredCatalog, null)))
+        .stream()
+        .map(AirbyteRecordMessage::getData).collect(Collectors.toSet());
+
+    configuredCatalog.getStreams().forEach(c -> c.setSyncMode(SyncMode.INCREMENTAL));
+    Set<JsonNode> dataFromDebeziumSnapshot =
+        extractRecordMessages(AutoCloseableIterators.toListAndClose(source.read(config, configuredCatalog, null)))
+            .stream()
+            .map(airbyteRecordMessage -> {
+              JsonNode data = airbyteRecordMessage.getData();
+              removeCDCColumns((ObjectNode) data);
+              /**
+               * Debezium reads TINYINT (expect for TINYINT(1)) as IntNode while FullRefresh reads it as Short Ref
+               * : {@link io.airbyte.db.jdbc.JdbcUtils#setJsonField(java.sql.ResultSet, int, ObjectNode)} -> case
+               * TINYINT, SMALLINT -> o.put(columnName, r.getShort(i));
+               */
+              ((ObjectNode) data)
+                  .put("tiny_int_two_col", (short) data.get("tiny_int_two_col").asInt());
+              return data;
+            })
+            .collect(Collectors.toSet());
+
+    assertEquals(dataFromFullRefresh, originalData);
+    assertEquals(dataFromFullRefresh, dataFromDebeziumSnapshot);
+  }
+
+  private void setupForComparisonBetweenFullRefreshAndCDCSnapshot(ImmutableList<JsonNode> data) {
+    executeQuery("CREATE DATABASE " + "test_schema" + ";");
+    executeQuery(String.format(
+        "CREATE TABLE %s.%s(%s INTEGER, %s Boolean, %s TINYINT(1), %s TINYINT(2), PRIMARY KEY (%s));",
+        "test_schema", "table_with_tiny_int", "id", "bool_col", "tiny_int_one_col",
+        "tiny_int_two_col", "id"));
+
+    executeQuery(String
```

Comment:
nitpick: are these two queries identical? if yes, probably better to loop over `data` rather than copy/pasting them?

Reply:
done